docs/_config.yml

```diff
@@ -26,3 +26,15 @@ SCALA_VERSION: "2.12.10"
 MESOS_VERSION: 1.0.0
 SPARK_ISSUE_TRACKER_URL: https://issues.apache.org/jira/browse/SPARK
 SPARK_GITHUB_URL: https://github.com/apache/spark
+# Before a new release, we should apply a new `apiKey` for the new Spark documentation
```

Comment:
An alternative way is, we apply a new API key for the previous release instead. For example, 
we are going to release 3.1.1 and the current release is 3.1.0. We apply a new API key for https://spark.apache.org/docs/3.1.0 and publish it. After that, we publish 3.1.1 with the default API key which points to the latest doc.

Reply:
Yeah